src/Octoshift/RetryPolicy.cs

```diff
@@ -29,7 +29,7 @@ public async Task<T> HttpRetry<T>(Func<Task<T>> func, Func<HttpRequestException,
             return await policy.ExecuteAsync(func);
         }
 
-        public async Task<PolicyResult<T>> RetryOnResult<T>(Func<Task<T>> func, T resultFilter, string retryLogMessage = null)
+        public async Task<PolicyResult<T>> RetryOnResult<T>(Func<Task<T>> func, Func<T, bool> resultFilter, string retryLogMessage = null)
```

Comment:
This is more flexible and allows you to define whatever filter you want, unlike the old way.

Reply:
I was forced the change this by the type checker. It might be best to pair so we can work on this together!